aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.spec.move

```diff
@@ -1,73 +1,106 @@
 spec aptos_framework::aggregator_v2 {
-    spec create_aggregator {
-        // TODO: temporary mockup.
-        pragma opaque;
+
+    spec Aggregator {
+        pragma intrinsic;
     }
 
-    spec create_unbounded_aggregator {
-        // TODO: temporary mockup.
-        pragma opaque;
+    spec create_aggregator<IntElement: copy + drop>(max_value: IntElement): Aggregator<IntElement> {
+        pragma intrinsic;
     }
 
-    spec try_add {
-        // TODO: temporary mockup.
-        pragma opaque;
+    spec create_unbounded_aggregator<IntElement: copy + drop>(): Aggregator<IntElement> {
+        pragma intrinsic;
     }
 
-    spec try_sub {
-        // TODO: temporary mockup.
-        pragma opaque;
+    spec try_add<IntElement>(aggregator: &mut Aggregator<IntElement>, value: IntElement): bool {
+        pragma intrinsic;
     }
 
-    spec is_at_least_impl {
-        // TODO: temporary mockup.
-        pragma opaque;
+    spec add<IntElement>(aggregator: &mut Aggregator<IntElement>, value: IntElement) {
+        pragma intrinsic;
     }
 
-    spec read {
-        // TODO: temporary mockup.
-        pragma opaque;
+    spec try_sub<IntElement>(aggregator: &mut Aggregator<IntElement>, value: IntElement): bool {
+        pragma intrinsic;
     }
 
-    spec snapshot {
-        // TODO: temporary mockup.
+    spec sub<IntElement>(aggregator: &mut Aggregator<IntElement>, value: IntElement) {
+        pragma intrinsic;
+    }
+
+    spec is_at_least_impl<IntElement>(aggregator: &Aggregator<IntElement>, min_amount: IntElement): bool {
+        pragma intrinsic;
+    }
+
+    spec read<IntElement>(aggregator: &Aggregator<IntElement>): IntElement {
+        pragma intrinsic;
+    }
+
+    spec snapshot<IntElement>(aggregator: &Aggregator<IntElement>): AggregatorSnapshot<IntElement> {
         pragma opaque;
+        include AbortsIfIntElement<IntElement>;
+        ensures [abstract] result.value == spec_get_value(aggregator);
```

Comment:
Question, why here we need to call a spec native for aggregator, and not aggregator.value? Is it because the spec is intrinsic? Cause technically for sub for example you can say: aggregator.value = old(aggregator).value - amount if no overflow, etc. or does the spec not support these kind of things because of generic types?

Reply:
yeah, when the spec of a struct is intrinsic, we cannot directly get access to its field. The reason of using intrinsics for spec is because generic types do not support `+` and `-`.

Reply:
Got it, thanks?